sprint0/sprint0/Collision/LinkProjectileCollisionHandler.cs

```diff
@@ -10,13 +9,20 @@ public LinkProjectileCollisionHandler()
 
         public void HandleCollision(IPlayer link, IProjectile projectile, Direction side)
         {
-            if (projectile.Shooter != null && !(projectile.Shooter is IPlayer))
-                link.TakeDamage(side, projectile.Damage);
-            else if (projectile.Shooter is IPlayer && projectile is Boomerang boomerang && boomerang.CanBeCaught)
+            if (projectile.Shooter is IPlayer && projectile is Boomerang boomerang && boomerang.CanBeCaught)
             {
                 boomerang.Perish();
                 link.ReceiveItem(1, PlayerItems.Boomerang);
             }
+            else if (!(projectile.Shooter is IPlayer))
+            {
+                projectile.RegisterHit();
+                if (projectile.Shooter != null && !(projectile.Shooter is IPlayer) &&
```

Comment:
isn't the !(projectile.Shooter is Player) redundant here since the condition is also in the else if